gpt_index/data_structs/data_structs_v2.py

```diff
@@ -201,9 +200,10 @@ def add_node(
     def delete(self, doc_id: str) -> None:
         """Delete a Node."""
         if doc_id not in self.doc_id_dict:
-            raise ValueError("doc_id not found in doc_id_dict")
```

Comment:
out of curiosity, does this primarily apply to vector indexes that aren't the simple vector index?

Reply:
@jerryjliu pretty much! But I didn't see the harm in applying it to all vector indexes 🤔

Reply:
Is there a better option you were thinking of? With the 3rd party vector stores, the documents are sometimes in the remote client (which we call delete for after the index struct), and sometimes also in the index, so we still want to delete it if it's there

Reply:
yeah, that makes sense to me. there is a bit of messiness in deciding what data goes into vector store and what doesn't, will take a look at cleaning it up